pkg/kubelet/lifecycle/predicate.go

```diff
@@ -221,19 +225,16 @@ func (e *PredicateFailureError) GetReason() string {
 }
 
 // GeneralPredicates checks a group of predicates that the kubelet cares about.
-func GeneralPredicates(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo) ([]PredicateFailureReason, error) {
+func GeneralPredicates(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo, nodeCapacity v1.ResourceList) ([]PredicateFailureReason, error) {
 	if nodeInfo.Node() == nil {
 		return nil, fmt.Errorf("node not found")
 	}
 
-	var reasons []PredicateFailureReason
-	for _, r := range noderesources.Fits(pod, nodeInfo) {
-		reasons = append(reasons, &InsufficientResourceError{
-			ResourceName: r.ResourceName,
-			Requested:    r.Requested,
-			Used:         r.Used,
-			Capacity:     r.Capacity,
-		})
+	// retry if ephemeral storage doesn't fit for not initialized
+	reasons, retry := predicatesNodeResources(pod, nodeInfo, nodeCapacity)
+	if retry {
+		time.Sleep(5 * time.Second)
```

Comment:
If we're going to take this approach, shouldn't we retry indefinitely (with a reasonable timeout) rather than just wait 5 seconds once? I'm not sure a ResourceEphemeralStorager value of 0 is ever a valid value for a working cluster, it should always indicate that we can't proceed, right?

Reply:
If we retried indefinitely, a retry log should be necessary.

Reply:
Updated. Would you take a look? @jackfrancis